seqr/views/apis/data_manager_api.py

```diff
@@ -387,14 +387,68 @@ def load_rna_seq_sample_data(request, sample_guid):
         data_by_gene = json.loads(row.split('\t\t')[1])
 
     model_cls = RNA_DATA_TYPE_CONFIGS[data_type]['model_class']
-    models = model_cls.objects.bulk_create([model_cls(sample=sample, **data) for data in data_by_gene.values()])
-    logger.info(f'create {len(models)} {model_cls.__name__}', request.user, db_update={
-        'dbEntity': model_cls.__name__, 'numEntities': len(models), 'parentEntityIds': [sample_guid], 'updateType': 'bulk_create',
-    })
+    model_cls.bulk_create(request.user, [model_cls(sample=sample, **data) for data in data_by_gene.values()], parent='sample')
 
     return create_json_response({'success': True})
 
 
+def _log_append_info(user, info, message):
+    info.append(message)
+    logger.info(message, user)
+
+
+def _load_phenotype_prioritization(file_path, user):
+    tool, data_by_project_sample_id = load_phenotype_prioritization_data_file(file_path)
+
+    info = []
+    _log_append_info(user, info, f'Parsed {tool.upper()} data for project(s): {", ".join(data_by_project_sample_id.keys())}')
+
+    all_records = []
+    to_delete = None
+    for project, records_by_sample in data_by_project_sample_id.items():
+        indivs = Individual.objects.filter(family__project__name=project, individual_id__in=records_by_sample.keys())
```

Comment:
project names are not unique. Use the project model itself - `family__project=project`

Reply:
Oh, you are right. So, shouldn't the `project` column be the project name? When two projects with the same name have individuals with identical IDs, it will be trouble.

Reply:
discussed offline, need to first get the list of projects, check permission, see that the projects actually exist, and validate that there is only one project per name